### PR TITLE
Created Reaper Proto

### PR DESCRIPTION
**Pull request by brianmodel**
Created the proto file for the reaper, and its associated bazel build file.

### CODE REVIEW

File: proto/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+go_proto_library(
+    name = "reaper_go_proto",
+    importpath = "github.com/googleinterns/cloudai-gcp-test-resource-reaper",
+    proto = ":reaper_proto",
+    visibility = ["//visibility:public"],
+)
+
+go_library(
```
Comment (supercclank) [inline proto/BUILD.bazel:18]: Was this created by gazelle? Is it needed or can it be removed?

File: proto/reaperconfig.proto [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+*/
+message ReaperConfig {
+    // List of resources to watch
+    repeated ResourceConfig resources = 1;
+
+    // Frequency to run reaper in cron time string format
+    string schedule = 2;
+    
+    // Regex of names of resources to exclude
+    string skipFilter = 3;
```
Comment (supercclank) [inline proto/reaperconfig.proto:42]: Please follow the guidelines in https://developers.google.com/protocol-buffers/docs/style

Messages should be CamelCase but fields should be underscore_separated_names

File: proto/reaperconfig.proto [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+/*
+A resource config describes a group of cloud resource and their time to live
+(TTL). A resource will be deleted once it has been alive for longer than its
+TTL. Note that if a resource is caught by both the the name filter and skip
+filter, the resource will be excluded (i.e. not monitored).
+*/
+message ResourceConfig {
+    // Type of GCP resource
```
Comment (supercclank) [inline proto/reaperconfig.proto:58]: Please add punctuation. Apply throughout.

File: proto/reaperconfig.proto [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+filter will be excluded, even if it matches a name filter in a resource config.
+*/
+message ReaperConfig {
+    // List of resources to watch
+    repeated ResourceConfig resources = 1;
+
+    // Frequency to run reaper in cron time string format
+    string schedule = 2;
+    
+    // Regex of names of resources to exclude
```
Comment (supercclank) [inline proto/reaperconfig.proto:41]: In both cases where skipFilter is used please add an inline comment that skips win over matches

File: proto/reaperconfig.proto [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    string nameFilter = 2;
+    
+    // Regex of names of resources to exclude
+    string skipFilter = 3;
+    
+    // List of which GCP zones to search
+    repeated string zones = 4;
+    
+    // Time to live of resources described in cron time string format
+    string TTL = 5;
```
Comment (supercclank) [inline proto/reaperconfig.proto:71]: Let's make this lowercase.

File: proto/BUILD.bazel (hunk 1 above)
Comment (brianmodel) [reply proto/BUILD.bazel:18]: It is needed when referencing the proto in other files

File: proto/reaperconfig.proto (hunk 2 above)
Comment (brianmodel) [reply proto/reaperconfig.proto]: Fixed

File: proto/reaperconfig.proto (hunk 3 above)
Comment (brianmodel) [reply proto/reaperconfig.proto]: Fixed

File: proto/reaperconfig.proto (hunk 4 above)
Comment (brianmodel) [reply proto/reaperconfig.proto]: Added

File: proto/reaperconfig.proto (hunk 5 above)
Comment (brianmodel) [reply proto/reaperconfig.proto]: Fixed

**Pull request merged**